python-frontend/src/main/java/org/sonar/python/types/TypeShed.java

```diff
@@ -120,39 +125,47 @@ private static void setDeclaredReturnType(Symbol symbol, FunctionDef functionDef
     return typingModuleSymbols;
   }
 
-  private static Map<String, Symbol> getModuleSymbols(String resourcePath, String moduleName) {
+  static Set<Symbol> typingExtensionsSymbols(Map<String, Set<Symbol>> typingSymbols) {
+    Map<String, Symbol> typingExtensionSymbols = getModuleSymbols("typeshed/third_party/2and3/typing_extensions.pyi", TYPING_EXTENSIONS,
+      typingSymbols);
+    return typingExtensionSymbols.values().stream().filter(s -> s.fullyQualifiedName().startsWith(TYPING_EXTENSIONS)).collect(Collectors.toSet());
+  }
+
+  private static Map<String, Symbol> getModuleSymbols(String resourcePath, String moduleName, Map<String, Set<Symbol>> initialSymbols) {
     InputStream resource = TypeShed.class.getResourceAsStream(resourcePath);
     PythonFile file = new TypeShedPythonFile(resource, moduleName);
     AstNode astNode = PythonParser.create().parse(file.content());
     FileInput fileInput = new PythonTreeMaker().fileInput(astNode);
-    new SymbolTableBuilder("", file, Collections.emptyMap()).visitFileInput(fileInput);
+    new SymbolTableBuilder("", file, initialSymbols).visitFileInput(fileInput);
     return fileInput.globalVariables().stream()
       .map(symbol -> {
         ((SymbolImpl) symbol).removeUsages();
         return symbol;
       })
-      .filter(s -> s.fullyQualifiedName() != null)
+      .filter(s -> s.fullyQualifiedName() != null && s.fullyQualifiedName().startsWith(moduleName))
       .collect(Collectors.toMap(Symbol::fullyQualifiedName, Function.identity()));
   }
 
-  public static Set<Symbol> standardLibrarySymbols(String stdlibModuleName) {
-    if (!TypeShed.standardLibrarySymbols.containsKey(stdlibModuleName)) {
-      Map<String, Symbol> result = readTypeShedSymbols("typeshed/stdlib/2and3/" + stdlibModuleName + ".pyi", stdlibModuleName);
+  public static Set<Symbol> typeShedSymbolsForModule(String stdlibModuleName) {
```

Comment:
Minor comment: I think we don't need to prefix the method name with "typeshed" as it's part of TypeShed class.

Reply:
If that's okay for you, I'll keep the name here, as the next PR I'm preparing includes further refactoring of this method, I'll take the opportunity to review the names consistency at the same time.